### PR TITLE
Add block mask utility support for batches and heads > 1

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #130250
* __->__ #130227

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: aten/src/ATen/native/IndexingUtils.h [hunk 1]
```diff
@@ -48,18 +48,12 @@ static C10_UNUSED std::vector<Tensor> expandTensors(const Tensor & self, IOptTen
   return result;
 }
 
-static C10_UNUSED void checkIndexTensorTypes(IOptTensorListRef indices, bool allow_int=false) {
+static C10_UNUSED void checkIndexTensorTypes(IOptTensorListRef indices) {
```
Comment (Chillee) [inline aten/src/ATen/native/IndexingUtils.h:51]: This is only used in 3 places, and all of them have `allow_int` set to True or *should* have it set to True.

**Review by yanboliang — APPROVED** [review]

**Comment by izaitsevfb** [comment]
@pytorchbot revert -m "breaks internal builds, please see D59498662" -c ghfirst

```
 error: too many arguments to function call, expected single argument 'indices', have 2 arguments
  at::native::checkIndexTensorTypes(indices, /*allow_int*/ true);
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                        ^~~~
buck-out/v2/gen/fbcode/795a8ff4e6618da8/caffe2/__aten-headers-cpu__/buck-headers/ATen/native/IndexingUtils.h:51:24: note: 'checkIndexTensorTypes' declared here
static C10_UNUSED void checkIndexTensorTypes(IOptTensorListRef indices) {
```

**Pull request closed without merging**